packages/plugin-legacy/src/index.ts

```diff
@@ -630,6 +630,7 @@ export async function detectPolyfills(
     ast: true,
     babelrc: false,
     configFile: false,
+    compact: true,
```

Comment:
Maybe it should be set to `false` instead? Seems like `true` adds some extra work when transforming which shouldn't be needed here. I think `false` should suppress it too reading the [source code](https://github.com/babel/babel/blob/257c06457f3e219d8fe90586937354b566a99476/packages/babel-generator/src/index.ts#L102-L111).

Reply:
You are right! Confirmed locally that `false` suppresses the warning.